.github/workflows/build.yml

```diff
@@ -112,6 +120,24 @@ jobs:
         run: ./scripts/fix-home stack install
         shell: bash
 
-      - name: Run tests
+      - name: Run tests (PureScript < 0.15.0)
         run: ./scripts/fix-home stack test
         shell: bash
+
+      - name: Download PureScript 0.15.0
+        shell: bash
+        env:
+          DL_FILE: ${{ matrix.purs_0_15_download_file }}
+          BIN_FILE: ${{ matrix.purs_0_15_binary_file }}
+        run: |
+          DIR="$HOME/bin/purescript-$PS_0_15_0_VERSION"
+          mkdir -p "$DIR"
+          curl -o "$DIR/purescript.tar.gz" -L "https://github.com/purescript/purescript/releases/download/$PS_0_15_0_VERSION/$DL_FILE.tar.gz"
```

Comment:
I guess we have to do this because the alphas are not on NPM, right? What if we'd publish those (but not make them install by default of course) on NPM as well?

Reply:
We haven't done this in the past. I'd bring it up with the core team.

Reply:
Could you clarify why that's an issue?

Reply:
It's churn, it's error prone, and I assume it's likely something that I'll have to take care of myself.
So trying to minimize the churn on this right now is less work for future me, unless you or someone else involved in this PR would like to do the update once 0.15 is out.
My opinion is that getting the release out on npm is strictly less work, but I might be missing something.

Reply:
Sorry for taking a few days before responding to this. I've been focusing on the release script.

> My opinion is that getting the release out on npm is strictly less work, but I might be missing something.

After thinking about this for a while, I think I agree with you. Since CI currently builds, I'd say the above statement is not necessarily true _in the short term_. However, I do think it's true in the _long-term_ when updating tools to work on a new breaking change compiler release. For example, it took me a while both in `pulp` and here to figure out what commands I have to run to correctly install the binary in CI. Delegating that to `npm` is a lot easier rather than figuring this out every time we do a breaking change PS release since debugging CI is a pain. Moreover, by having pre-releases available via `npm`, more people are likely to try it out and find potential problems with it before the core team does.

The only concern I have with this is if a release is done improperly and the 'next' release points to the pre-release rather than the last stable one. I don't have much experience with publishing to NPM, so how would we do that?

> It's churn, it's error prone, and I assume it's likely something that I'll have to take care of myself.
> So trying to minimize the churn on this right now is less work for future me, unless you or someone else involved in this PR would like to do the update once 0.15 is out.

Even if we did publish the pre-releases to NPM, there's still a lot more work than just what this PR entails before spago (and all of its CI) is fully updated to 0.15.0. Since you're already busy with other things, I can see how this just feels like another TODO in an already busy life. @sigma-andex, would you be willing to work with me to get this done?

Once `0.15.0` is actually released, I think there are a couple of changes we would need to make:

First, we would replace the custom download script with just `npm install -g purescript@0.15.0`. In other words, this...:
```
- name: Download PureScript 0.15.0
  shell: bash
  env:
    DL_FILE: ${{ matrix.purs_download_file }}
    BIN_FILE: ${{ matrix.purs_binary_file }}
  run: |
    DIR="$HOME/bin/purescript-$PS_0_15_0_VERSION"
    mkdir -p "$DIR"
    curl -o "$DIR/purescript.tar.gz" -L "https://github.com/purescript/purescript/releases/download/$PS_0_15_0_VERSION/$DL_FILE.tar.gz"
    tar -xvf "$DIR/purescript.tar.gz" -C "$DIR" --strip-components 1 "purescript/$BIN_FILE"
    chmod a+x "$DIR/$BIN_FILE"

- name: Run tests (PureScript >= 0.15.0)
  shell: bash
  run: |
    PATH="$HOME/bin/purescript-$PS_0_15_0_VERSION:$PATH"
    ./scripts/fix-home stack test --ta "--match purs-0.15"
```
would be updated to this:
```
- name: Download PureScript 0.15.0
  shell: bash
  run: npm install -g purescript@0.15.0

- name: Run tests (PureScript >= 0.15.0)
  shell: bash
  run: ./scripts/fix-home stack test --ta "--match purs-0.15"
```

We'd also drop the `env.PS_0_15_0_VERSION` and `jobs.build.strategy.matrix.include.[purs_download_file/purs_binary_file]` stuff.

Second, there are a lot of tests not being run with the `0.15.0` alpha version because we don't have a released 0.15.0 package set yet. So, the tests may also need to be updated.

Third, some packages (e.g. `simple-json`) used throughout the test may take a while to get into the package set. So, even if we have a released 0.15.0 package set, either someone would need to update that in a PR or change the tests to stop using those packages. 

Fourth, there may be some tests we want to continue running when using `0.14.0`  but not `0.15.0` or vice versa (just hypothesizing here).

Reply:
> I don't have much experience with publishing to NPM, so how would we do that?

Answering my own question, it looks like we follow [these instructions](https://gist.github.com/schmich/d04acc4b02b45e489f329cfdf3280a3f).

Reply:
I've opened https://github.com/purescript/purescript/issues/4294 to discuss the publish to NPM idea.

Reply:
@JordanMartinez sorry I somehow missed this threat here. Sure, can help you with this task. Let me know how I can help you.

Reply:
I've updated the code to use `npm i -g purescript@next`. Just waiting on CI to build now.